src/main/resources/avro/referencemethods.avdl

```diff
@@ -113,9 +113,9 @@ as JSON.
 */
 record SearchReferencesRequest {
   /**
-  If present, return only references which belong to this reference set.
+  The ReferenceSet to search.
   */
-  union { null, string } referenceSetId = null;
```

Comment:
Pre-graph changes: it wasn't required but it wasn't disallowed. If a repository serves reference sets hg19 and GRCh37, it would be up to the implementation whether chr4 (MD5 is identical in both assemblies) is represented as one or two references. The former is easier to implement and might be required in some cases, e.g. if ACLs are involved, but the latter may be conceptually cleaner. But yes, it should probably be made explicit in the docs that this is allowed.

Things have also changed with the graph PR. Now you can nest reference sets within a reference set (I missed that when I was reviewing the changes). I guess that makes the semantics of this field slightly more complicated and I'm not sure how it plays with many : many.

If reference sets are still many:many with references after the graph changes, then I'd definitely prefer this line be reverted.